src/nmodl/solve.cpp

```diff
@@ -125,21 +136,11 @@ void solvhandler() {
         qsol = ITM(lq);
         lq = lq->next;
         method = SYM(lq);
-        cvodemethod_ = 0;
-        if (method && strcmp(method->name, "after_cvode") == 0) {
-            method = (Symbol*) 0;
-            lq->element.sym = (Symbol*) 0;
-            cvodemethod_ = 1;
-        }
-        if (method && strcmp(method->name, "cvode_t") == 0) {
-            method = (Symbol*) 0;
-            lq->element.sym = (Symbol*) 0;
-            cvodemethod_ = 2;
-        }
-        if (method && strcmp(method->name, "cvode_v") == 0) {
-            method = (Symbol*) 0;
-            lq->element.sym = (Symbol*) 0;
-            cvodemethod_ = 3;
+        cvodemethod_ = CVodeMethod::nomethod;
+        if (method && cvode_method_map.count(method->name)) {
+            cvodemethod_ = cvode_method_map.at(method->name);
```

Comment:
```suggestion
        cvodemethod_ = deduce_cvode_method(method);
        if (cvodemethod_ != CVodeMethod::none) {
```
You can implement `deduce_cvode_method` in two stages, first convert `method` to a `char*`. Then convert the `char*` to enum, returning `CVodeMethod::none` if the string isn't one of the supported names for the cvode methods.